src/utils.jl

```diff
@@ -0,0 +1,22 @@
+function custom_solver() end
+
+function solve_block(
```

Comment:
I would suggest renaming this function as `optimize_block!` (the `!` specifying we update the `block` inplace)